editor/editor_node.cpp

```diff
@@ -1983,6 +1983,8 @@ void EditorNode::_dialog_action(String p_file) {
 				if (scene_idx != -1) {
 					_discard_changes();
 				}
+
+				editor_data.set_scene_path(scene_idx, p_file); // Update the path of the edited scene.
```

Comment:
This is a fix for the case of action "Add atlas source". See the video for steps to reproduce.